### ISSUE TITLE
OBS virtual camera crashing applications when resolution changes

### ISSUE THREAD

**Issue by notr1ch** [obsproject/obs-studio#7529]
### Operating System Info

Windows 11

### OBS Studio Version

28.0.3

### OBS Studio Log URL

N/A

### Expected Behavior

No crashing.

### Current Behavior

Crashing when resolutions change.

### Steps to Reproduce

1. Set OBS to > 1080p output.
2. With the output in OBS inactive, go in to Settings -> Voice & Video in Discord
3. Select OBS Virtual Camera, and click "Test Video"
4. Start the virtual camera output in OBS
5. Switch to another settings tab in Discord (such as Accessibility)
6. Switch back to Voice & Video, and click Test Video again
7. Discord will crash

### Anything else we should know?

This is a tracking issue for a known virtualcam bug that can manifest as applications crashing when they request a different resolution from the virtualcam. There are _many_ possible reasons why an application might crash when using virtualcam - this issue is ONLY for instances where the application requests a different resolution than OBS is outputting. This generally only crashes at > 1080p resolutions due to memory overwriting (1080p being the default buffer size).

For other virtualcam application crashes that do NOT involve resolution changes, please file a new issue. If you aren't sure, please open a new issue and we will triage as necessary. Previous virtualcam issue reports became impossible to follow with many different bugs all being reported in a single issue.

**Comment by notr1ch** [comment]
Here is a potential fix for this specific issue, please test this and let me know if this solves your virtual camera issues.

Download: [notr1ch-vcam-resize-721e4fbe.zip](https://github.com/obsproject/obs-studio/files/9767254/notr1ch-vcam-resize-721e4fbe.zip)

**Instructions:**
Make sure no software is trying to access the virtual camera. Download and extract the files into `C:\Program Files\obs-studio\data\obs-plugins\win-dshow` (or wherever you installed OBS), overwriting the existing files. You do not need to use the uninstall / install scripts. If this fix ends up making things worse, you can use Help / Check File Integrity to revert to the current released versions.

(Source for anyone interested: https://github.com/notr1ch/obs-studio/commit/721e4fbea82b2657b58df2015b92bba6a414c8cd)

**Comment by notr1ch** [comment]
A fix should for this should be shipping in 28.1.

https://github.com/obsproject/obs-studio/commit/feda64e9c618ed217a6549f3457960a54ebef207

**Comment by ticpu** [comment]
> Here is a potential fix for this specific issue, please test this and let me know if this solves your virtual camera issues.
> Download: [notr1ch-vcam-resize-721e4fbe.zip](https://github.com/obsproject/obs-studio/files/9767254/notr1ch-vcam-resize-721e4fbe.zip)

This version works perfectly on my OBS 29.0.2 which seem to have the same (or worst) issue than before.  Turning the Virtual Camera off and on twice automatically crashes Discord, whatever or not the resolution is >1080p.  I guess I'll open a new issue on the next OBS update if it happens again.

**Comment by ChaoticHolon** [comment]
This needs to be re-opened! It seems to be even worse than it was before, as the previous poster stated. This fix didn't fully work for me, however on OBS 29.1.1.

The fix does fix toggling the virtual camera on and off (in OBS), but if you toggle the camera off and then back on in Discord, it crashes instantly.

**Comment by Fenrirthviti** [comment]
While the issue might seem similar, this specific case was already corrected. If you are having other issues with crashing, please open a new case with sufficient information to reproduce the issue, and we can take a closer look.

**Comment by ChaoticHolon** [comment]
It is not, I am fairly certain this is a regression, as it was fixed for a bit and then seemed to return. It is a lot of work to open a new case/ticket, so that may or may not happen, but I wish you guys would investigate this anyways.

**Comment by Fenrirthviti** [comment]
I have already retested this, and it does not crash anymore given the information and steps to reproduce in this thread. If this is still crashing, we need a new issue opened, with new steps to reproduce before we can look further. I'm not dismissing that there might be an issue, but this specific issue was already fixed, so whatever is happening is likely something new, and not related to this.

**Comment by superconvert** [comment]
> I have already retested this, and it does not crash anymore given the information and steps to reproduce in this thread. If this is still crashing, we need a new issue opened, with new steps to reproduce before we can look further. I'm not dismissing that there might be an issue, but this specific issue was already fixed, so whatever is happening is likely something new, and not related to this.

https://github.com/obsproject/obs-studio/issues/9119
